deployment/environment/crib/lane_config.go

```diff
@@ -0,0 +1,520 @@
+package crib
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"sort"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+
+	"github.com/AlekSi/pointer"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	selectors "github.com/smartcontractkit/chain-selectors"
+
+	cldf_chain "github.com/smartcontractkit/chainlink-deployments-framework/chain"
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/stateview"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/stateview/evm"
+	solState "github.com/smartcontractkit/chainlink/deployment/ccip/shared/stateview/solana"
+)
+
+// LaneConfig represents a unidirectional lane from source to destination
+type LaneConfig struct {
+	SourceChain      uint64 `toml:",omitempty"`
+	DestinationChain uint64 `toml:",omitempty"`
+}
+
+// LaneConfiguration defines how lanes should be configured for the load test
+type LaneConfiguration struct {
+	// Mode determines how lanes are configured
+	// "any-to-any" - traditional full mesh (default)
+	// "random-lanes" - generate random lanes based on count
+	Mode *string `toml:",omitempty"`
+
+	// NumLanes - number of random lanes to generate when Mode is "random-lanes"
+	NumLanes *int `toml:",omitempty"`
+
+	// EnsureBidirectional - when true, ensures that for every A->B lane, there's also a B->A lane
+	EnsureBidirectional *bool `toml:",omitempty"`
+
+	// Internal fields for caching and deterministic generation
+	generatedLanes []LaneConfig `toml:"-"` // Cache for generated lanes
+}
+
+const (
+	LaneModeAnyToAny    = "any-to-any"
+	LaneModeRandomLanes = "random-lanes"
+)
+
+func (lc *LaneConfiguration) Validate(e *cldf.Environment) error {
+	if lc == nil {
```

Comment:
ditto, it would be better to make that explicit and error when config is not present